build_projects/shared-build-targets-utils/Utils/Crossgen.cs

```diff
@@ -198,6 +198,9 @@ public void CrossgenDirectory(string sharedFxPath, string pathToAssemblies)
                 { "COMPlus_PartialNGen", "0" }
             };
 
+            // Form the dynamic path that would not collide if another instance of this is running.
+            string basePath = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
```

Comment:
Currently, it gets generated in the intermediates folder. I have added cleanup of the folder once the crossgen is complete.